File: Server/MessageAppServer/Model/Chat/Room.cs
using System.ComponentModel.DataAnnotations;
using System.Security.Cryptography;
using System.Text;

namespace Server.Model.Chat;

public class Room
{
    [Key]
    public Guid RoomId { get; private set; }
    public string RoomName { get; init; }
    public string Password { get; private set; }
    
    public Room() {}
    
    public Room(string roomName, string password)
    {
        RoomId = Guid.NewGuid();
        RoomName = roomName;
        Password = HashPassword(password);
    }

    public void SetRoomIdForTests(string id)
    {
        RoomId = new Guid(id);
    }

    private static string HashPassword(string password)
    {
        using var sha256 = SHA256.Create();
        var hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
            
        return BitConverter.ToString(hashedBytes).Replace("-", "").ToLower();
    }

    public bool PasswordMatch(string enteredPassword)
    {
        var enteredPasswordHash = HashPassword(enteredPassword);
        return Password == enteredPasswordHash;
    }

    public void ChangePassword(string password)
    {
        Password = HashPassword(password);
    }
}

File: Java_Server/src/main/java/org/example/java_server/Models/Chat/Room.java
package org.example.java_server.Models.Chat;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.UUID;

public class Room {
    private UUID RoomId;
    private String RoomName;
    private String Password;

    public Room(String roomName, String password) {
        this.RoomId = UUID.randomUUID();
        this.RoomName = roomName;
        this.Password = hashPassword(password);
    }

    public UUID getRoomId() {
        return this.RoomId;
    }

    public String getRoomName() {
        return this.RoomName;
    }

    public static String hashPassword(String password) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hashedBytes = digest.digest(password.getBytes());
            return bytesToHex(hashedBytes);
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
            return null; // Handle error appropriately
        }
    }

    public static boolean passwordMatch(String enteredPassword, String storedHashedPassword) {
        String enteredPasswordHash = hashPassword(enteredPassword);
        return storedHashedPassword.equals(enteredPasswordHash);
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}
